src/algebra/module/linear_map.lean

```diff
@@ -212,10 +212,9 @@ variables (f : M₂ →ₗ[R] M₃) (g : M →ₗ[R] M₂)
 /-- Composition of two linear maps is a linear map -/
 def comp : M →ₗ[R] M₃ := ⟨f ∘ g, by simp, by simp⟩
 
-@[simp] lemma comp_apply (x : M) : f.comp g x = f (g x) := rfl
+lemma comp_apply (x : M) : f.comp g x = f (g x) := rfl
 
-@[norm_cast]
-lemma comp_coe : (f : M₂ → M₃) ∘ (g : M → M₂) = f.comp g := rfl
+@[simp, norm_cast] lemma coe_comp : (f.comp g : M → M₃) = f ∘ g := rfl
```

Comment:
```suggestion
@[simp, norm_cast] lemma coe_comp : ⇑(f.comp g) = f ∘ g := rfl
```

Reply:
Why is using the arrow preferred over an explicit type annotation?

Reply:
The main reasons are:
* It's shorter
* we do that elsewhere
* It matches what the goal view shows and what doc-gen shows

It doesn't actually make any difference to lean.

Reply:
I prefer the explicit type annotation basically because I might be browsing the code outside of an editor or even if I am in the editor, if I'm looking at this lemma I might be too lazy to click around to see what the coercion is.

(Ultimately I don't think this is a big deal, so Yury should just go with whichever he wants.)

Reply:
@eric-wieser "Arrows" version doesn't work with the current state of https://github.com/leanprover-community/lean/pull/557 (I hope, it will be fixed), so I'll leave it as is.